shared/src/simulation/activities/unload-vehicle.ts

```diff
@@ -0,0 +1,76 @@
+import { IsInt, IsUUID, Min } from 'class-validator';
+import { SimulatedRegion } from '../../models';
+import { getCreate } from '../../models/utils';
+import { uuid, UUID, uuidValidationOptions } from '../../utils';
+import { IsValue } from '../../utils/validators';
+import { terminateActivity } from '../utils/simulated-region';
+import { unloadVehicle } from '../utils/vehicle';
+import type {
+    SimulationActivity,
+    SimulationActivityState,
+} from './simulation-activity';
+
+export class UnloadVehicleActivityState implements SimulationActivityState {
+    @IsValue('unloadVehiclesActivity' as const)
+    public readonly type = 'unloadVehiclesActivity';
+
+    @IsUUID(4, uuidValidationOptions)
+    public readonly id: UUID = uuid();
+
+    @IsUUID(4, uuidValidationOptions)
+    public readonly vehicleId!: UUID;
+
+    @IsInt()
+    @Min(0)
+    public readonly startTime!: number;
+
+    @IsInt()
+    @Min(0)
+    public readonly duration!: number;
+
+    /**
+     * @deprecated Use {@link create} instead
+     */
+    constructor(
+        vehicleId: UUID,
+        startTime: number,
+        duration: number,
+        id?: UUID
+    ) {
+        this.vehicleId = vehicleId;
+        this.startTime = startTime;
+        this.duration = duration;
+        if (id) this.id = id;
```

Comment:
Currently not but for consistency with all other objects in the state I made it optional.

Reply:
In my point of view, `id` is just some property like `startTime` or `duration` that has to be set to create a valid instance. Therefore, I don't think its inconsistent to enforce this parameter to be set and enforcing it can prevent us from hard to find bugs with inconsistent ids.